flybywiresim/a32nx
# fix(misc): Change of default switch positions for RAM AIR, WX Mode, TCAS ALT RPTG and TCAS Threat

Fixes #3965 

## Summary of Changes
Fixes the default loading positions of Pushbutton RAM AIR, WX Radar Mode, TCAS ALT RPTG and TCAS threat selector by editing the FLT files.

RAM AIR - Changed from ON with no light to OFF with no light, guard remains on. (All FLT files)
A32NX_AIRCOND_RAMAIR_TOGGLE=0

WX Radar Mode - Changed from WX to WX+T. (All FLT files)
XMLVAR_A320_WEATHERRADAR_MODE = 1

TCAS ALT RPTG - Added code into the xml of the ATC panel to repair the original missing call from the sim variables, FLT files were already trying to set it to on, but this missing code wasn't there.
Used: https://github.com/flybywiresim/a32nx/blob/2a0dfd9b78a87127ee71438dc214c1d7286bf230/docs/i-vars.md as found by Saschl#9432

TCAS Threat Selector - Changed from THRT to ABV. (FLT files from Cold and Dark until Cruise)
TCAS Threat Selector - Changed from THRT to BLW. (FLT files from Cruise to Landing)
A32NX_SWITCH_TCAS_TRAFFIC_POSITION = 2 (for ABV)
A32NX_SWITCH_TCAS_TRAFFIC_POSITION = 3 (for BLW)

## Screenshots (if necessary)

## References

RAM AIR
![RamAir1](https://user-images.githubusercontent.com/74841560/111155809-daefdb00-858c-11eb-9191-c578a4dba7ce.png)
![RamAir2](https://user-images.githubusercontent.com/74841560/111155820-ddeacb80-858c-11eb-8de4-9129733b89b3.png)

WX Radar MODE
![WXRadar1](https://user-images.githubusercontent.com/74841560/111155838-e5aa7000-858c-11eb-91de-7c729e55798c.png)
![WXRadar2](https://user-images.githubusercontent.com/74841560/111155849-e9d68d80-858c-11eb-8f93-db82ddef25f3.png)

TCAS
![TCAS1](https://user-images.githubusercontent.com/74841560/111155880-f1963200-858c-11eb-83c7-8cf529b892b1.png)
![TCAS2](https://user-images.githubusercontent.com/74841560/111155885-f35ff580-858c-11eb-9779-aca8291b20ca.png)


## Additional context
RAM AIR is only on for emergency use only, and should not be on in normal operation.

WX Mode is not defined for selections in manuals, but only AS REQUIRED. However, to have a weather radar with a functioning Turbulence mode is expected to be left in that switch position by all crews, otherwise the effectiveness of the system is downgraded.

TCAS ALT RPTG, should be left on at all times to not allow the mistake of departing without the Mode C reporting altitude, is never turned off unless under unusual data corruption operations.

TCAS THRT, only explicitly told in Airline OMB (SOP manual) as has to be in ABV for cockpit setup. This is however taught as being ABV up until the cruise, as it shows targets you may be climbing into, and then switched to BLW when cruise established, as to allow the identification of aircraft below you for descent or possible emergency descent. This could also be substituted for ALL if we model off another airline operations, but either is better than it being left on THRT.

**Additional: This is my first real PR into the master, please be very cautious in case I have edited anything in by mistake, extra precaution required even for the small amount of changes done.**


Discord username (if different from GitHub): Shomas#2719


## Testing instructions
1) Load the aircraft in different situations, cold and dark, on runway, in cruise, airborne and approach.
2) Note position of overhead guarded switch RAM AIR, which should have a cover closed on it.
3) Open cover to check that RAM AIR pushbutton is in the OFF position.
4) Check WX radar panel that the mode WX+T is selected on the left hand dial.
5) Check TCAS panel that ALT RPTG is on.
5) Check TCAS panel that THRT switch, is in ABV if loaded on ground or in climb, and in BLW if loading in cruise or onwards to landing.

Notes: Early QA tests have shown that people aren’t aware of how to jump between different flight loading states, this can be done by creating a basic flight plan on the in game map and then using the toolbar at the top to jump to different stages of flight, such as taxi, runway, cruise etc etc 

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
